Update AddEventCommandParser and DeleteEventCommandParser

YuFeng0930 opened the pull request:
What have been done:
- Add AddEventCommandParser class and DeleteEventCommandParser class without deleting the AB3 AddCommandParser and DeleteCommandParser.
- The parsers' dependencies are with SocheduleParserUtil
- Restore the AboutUs table (which was delete by me last time).
Closes #52, closes #53.

YuFeng0930 committed:
Merge commit 'b400c0a7475eb639bae5abb3d8115a75d28fd04e'

* commit 'b400c0a7475eb639bae5abb3d8115a75d28fd04e': (26 commits)
  Update the the title and repository.
  Fix displaying of user guide on website
  Fix styling
  UserGuide.md: Update add event feature
  DeveloperGuide.md: Fix spelling
  DeveloperGuide.md: Standardise formatting; add glossary terms
  UserGuide.md: Standardise formatting
  AboutUs.md: Update format and name to fit conventions
  DeveloperGuide.md: Fix missing headers
  Rename profile photo to fit naming requirement
  Minor fix for updates on Developer Guide for adding event and listing event
  Update Developer Guide for adding event and listing event
  Update User Guide for adding event and listing event
  Update AboutUs.md
  Convert user photo to .png.
  Update Developer Guide to have use cases
  Update Developer Guide to contain user profile and value proposition
  Update User Guide
  Update About Us
  Update UserGuide with "Done" command
  ...

YuFeng0930 committed:
Add class: AddEventCommandParser

Add a AddEventCommandParser without deleting the AB3 AddCommandParser.
Restore the AboutUs table (which was delete by me last time).

YuFeng0930 committed:
Add class: DeleteEventCommandParser

Add a DeleteEventCommandParser without deleting the AB3 DeleteCommandParser.

ljhgabe reviewed (approved):
LGTM

ljhgabe commented on src/main/java/seedu/address/logic/parser/AddEventCommandParser.java:
```diff
@@ -0,0 +1,65 @@
+        Time startTime = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_STARTTIME).get());
+        Date endDate = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_ENDDATE).get());
+        Time endTime = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_ENDTIME).get());
+        Set<Category> categoryList = ParserUtil.parseTags(argMultimap.getAllValues(PREFIX_CATEGORY));
```
Maybe can change the parseTags here to a parseCategories? 🤔

YuFeng0930 replied:
Thanks for spotting the error:)

YuFeng0930 committed:
Fix the parser function names

YuFeng0930 committed:
Merge commit '3c7ceaf1eb21ddc997992152ab87ddb91cc180cb' into iteration-1.1-YF

* commit '3c7ceaf1eb21ddc997992152ab87ddb91cc180cb': (41 commits)
  Fix checkstyle error because of indentatons
  Uncomment all prefixes in CliSyntax
  Temporarily remove SocheduleParser to resolve build error
  Update ParserUtil
  Add SocheduleParser
  Update CliSyntax
  Event.java: Rename Event.category to Event.categories
  Task.java Fix dependencies
  Task.java: Rename Task.category to Task.categories
  Event.java: Update datatype of this.category to Set<Category>
  Task.java: Update datatype of this.category to Set<Category>
  Overload ModelManager.java with Sochedule functions; Remove SocModelManager.java
  UserPrefs.java: Update equals and toString methods
  Fix file extension
  add SocModelManager for task
  Integrate Sochedule into Model
  Fix checkstyle
  Fix compiling error
  Add Sochedule and underlying ReadOnlySochedule
  Add methods to model interface
  ...

YuFeng0930 committed:
Add classes: AddEventCommand, DeleteEventCommand

What has been done:
- Add two classes and change "Person" to "Event"
- Change the import (Name & Tag) in ParserUtil to fit the other parsers
- Add constant messages in seedu.address.commons.core.Messages

litone01 commented on src/main/java/seedu/address/logic/parser/ParserUtil.java:
```diff
@@ -11,12 +11,12 @@
 import seedu.address.logic.parser.exceptions.ParseException;
 import seedu.address.model.common.Category;
 import seedu.address.model.common.Date;
+import seedu.address.model.common.Name;
```
this should not be imported yet.

litone01 commented on src/main/java/seedu/address/logic/parser/ParserUtil.java:
```diff
@@ -11,12 +11,12 @@
 import seedu.address.model.common.Category;
 import seedu.address.model.common.Date;
+import seedu.address.model.common.Name;
+import seedu.address.model.common.Tag;
```
this should not be imported yet.

YuFeng0930 committed:
Merge commit '20b55511e89eafbe8e895b4336fbfeac65e7f238' into iteration-1.1-YF

* commit '20b55511e89eafbe8e895b4336fbfeac65e7f238':
  Add SocheduleParserUtil
  Add dummy SocheduleParser The current AddressBookParser does not accept new commands like addtask and add event parsing.

YuFeng0930 committed:
Change dependencies to SocheduleParserUtil

litone01 commented on src/main/java/seedu/address/logic/commands/AddEventCommand.java:
```diff
@@ -0,0 +1,67 @@
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_NAME + "John Doe "
+            + PREFIX_PHONE + "98765432 "
+            + PREFIX_EMAIL + "johnd@example.com "
```
are we gonna change the prefix_name and phone and email etc

YuFeng0930 committed:
Merge commit '10adea33378aaeb0134bb03a7198781b3c04b28a' into iteration-1.1-YF

* commit '10adea33378aaeb0134bb03a7198781b3c04b28a':
  Update the message in AddCommand
  Task.java: Add Completion Status support
  Fix checkstyle for the update
  Update the corresponding add_task, delete_task and done command
  Update parser package to include DoneTaskCommandParser and update DoneTaskCommand class
  Update parser package to include DeleteTaskCommandParser and update relevant DeleteTaskCommand
  Update parser package to include AddTaskCommandParser

# Conflicts:
#	src/main/java/seedu/address/commons/core/Messages.java
